server/tests/integration_test.rs

```diff
@@ -0,0 +1,765 @@
+        output
+            .read_line(&mut line)
+            .expect("Could not read line of process.");
+    }
+
+    return process;
+}
+
+/// Create a new Arrow Flight client to access the endpoints in the ModelarDB server.
+fn create_flight_service_client(
+    rt: &Runtime,
+    host: &str,
+    port: u16,
+) -> Result<FlightServiceClient<Channel>, Box<dyn Error>> {
+    let address = format!("grpc://{}:{}", host, port);
+
+    rt.block_on(async {
+        let fsc = FlightServiceClient::connect(address).await?;
+        Ok(fsc)
+    })
+}
+
+/// Create a normal table or model table with/without tags in the
+/// ModelarDB server with Arrow Flight using the `do_action()` SQL Parser.
+/// mode options: NormalTable/ModelTable/ModelTableNoTag
+fn create_table(
+    rt: &Runtime,
+    client: &mut FlightServiceClient<Channel>,
+    table_name: &str,
+    mode: &str
+) {
+    let cmd = match mode{
+        "NormalTable" => format!(
+            "CREATE TABLE {}(timestamp TIMESTAMP, values REAL, metadata REAL)",
+            table_name
+        ),
+        "ModelTable" => format!(
+            "CREATE MODEL TABLE {}(timestamp TIMESTAMP, value FIELD, tag TAG)",
+            table_name
+        ),
+        "ModelTableNoTag" => format!(
+            "CREATE MODEL TABLE {}(timestamp TIMESTAMP, value FIELD)",
+            table_name
+        ),
+        _ => panic!("No mode selected in create_table.")
+    };
+
+    let action = Action {
+        r#type: "CommandStatementUpdate".to_string(),
+        body: cmd.into_bytes(),
+    };
+
+    rt.block_on(async {
+        client.do_action(Request::new(action)).await.expect("Could not create model table.");
+    })
+}
+
+/// Create the FlightDescriptor needed to initialize the stream.
+fn create_flight_data_from_generated_messages(messages: Vec<RecordBatch>) -> Vec<FlightData> {
+
+    let flight_descriptor = FlightDescriptor::new_path(vec![TABLE_NAME.to_string()]);
+
+    let mut flight_data_vec = vec![FlightData {
+        flight_descriptor: Some(flight_descriptor),
+        data_header: vec![],
+        app_metadata: vec![],
+        data_body: vec![],
+    }];
+    for message in messages{
+        flight_data_vec.push((flight_data_from_arrow_batch(&message, &ipc::writer::IpcWriteOptions::default())).1);
+    }
+
+    return flight_data_vec;
+}
+
+/// Generate a [`RecordBatch`] with the current timestamp, a random value and an optional tag.
+fn generate_random_message(tag: Option<String>) -> RecordBatch {
+    let timestamp = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .expect("Cannot generate the time.")
+        .as_micros() as i64;
+
+    let value = rand::thread_rng().gen_range(0..100) as f32;
+
+    let mut message_schema = Schema::new(vec![
+        Field::new("timestamp", DataType::Timestamp(Millisecond, None), false),
+        Field::new("value", DataType::Float32, false),
+    ]);
+
+    if tag.clone().unwrap_or("".to_string()) == "" {
```

Comment:
Why not use `if let Some(tag)` here? it seems to be the best way to both check if the tag exists and extract it from the option.

Reply:
I believe that this was because it still passes the edge case of inserting an empty string instead of a `None`. Maybe that is a bit of a stretch, so I changed it as you suggested. If you think I should change it back, leave another comment, otherwise resolve it.